### DISCUSSION TITLE
Payload on pings

### DISCUSSION THREAD

**Discussion by bakhtiyarneyman** [enisdenjo/graphql-ws#575] [Q&A] [answered]
### Story

We are implementing a chat. User messages to the chat are mutations, and chat messages by other users are delivered back to the user via a subscription over graphql-ws. We'd like to know when the user is disconnected. For that we want to use keep-alive functionality. But there are additional transient metadata that we'd like to deliver to the server (e.g. "user is typing"). This would fit nicely into the optional `payload` field defined on the client Ping message as per GraphQL spec. 

https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md#ping 

We don't want to use mutations for that because these events are too frequent, and processing a separate request just for that is expensive.

### Acceptance criteria

- Client can deliver a payload to the server using pings.

**Comment by enisdenjo** [comment, marked as answer]
Yes, the ping and pong payloads is ideal for this scenario. As inspiration, you can advise the ["client usage with manual pings and pongs" recipe](https://the-guild.dev/graphql/ws/recipes#client-usage-with-manual-pings-and-pongs) for implementing a client side pinger and ["server usage with ws and subprotocol pings and pongs" recipe](https://the-guild.dev/graphql/ws/recipes#server-usage-with-ws-and-subprotocol-pings-and-pongs) for server side.

**Reply by bakhtiyarneyman** [reply]
Thanks! I'm not sure why I thought it wasn't implemented yet. Maybe I was looking at an old version.
